_rules/letter-spacing-not-important-24afc2.md

```diff
@@ -0,0 +1,161 @@
+---
+id: 24afc2
+name: Letter spacing in `style` attributes is not `!important`
+rule_type: atomic
+description: |
+	This rule checks that the `style` attribute is not used to prevent adjusting `letter- spacing` by using `!important`, except if it's at least `0.12` times the font size.
+accessibility_requirements:
+  wcag21:1.4.12: # Text Spacing (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Jey Nandakumar
+---
+
+## Applicability
+
+This rule applies to any HTML element that is [visible][], for which the `style` attribute [declares][declared] the [letter-spacing][] CSS property.
+
+## Expectation
+
+For the test target, one of the following is true:
+
+- the [cascaded][] value of its [letter-spacing][] property, specified via the `style` attribute, does not have the [important flag][]; or
```

Comment:
```suggestion
- the [cascaded][] value of its [letter-spacing][] property, if specified via the `style` attribute, does not have the [important flag][]; or
```

This applies to the other rules as well. Since the cascaded value may come from a declaration not in the `style` attribute, we're only interested in the important flag of the cascaded value _if_ it comes from a declaration in the `style` attribute.

Reply:
I don't think this is enough. 

```html
<html>
 	<style>
 		.noSpace {
 			letter-spacing: 0.01em !important;
 		}
 	</style>

 	<body>
 		<div class="noSpace" style="letter-spacing: 0.12em">
 			The toy brought back fond memories of being lost in the rain forest.
 		</div>
 	</body>
 </html>
```

We can't say anything about condition 1 because the cascaded value was not specified through the `style` attribute. Condition 2 is false. We need condition 1 to pass this example. I guess we need to use the declared value in condition 1 instead of the cascaded value.

Reply:
Good point!

> I guess we need to use the declared value in condition 1 instead of the cascaded value.

Then we're back to square 1 though 🙈 The issue is that there can be multiple declared values and we want the one that wins cascade, if any. Something like this might work to ensure totality:

> - the cascaded value of its `letter-spacing` property is either not specified through the `style` attribute or does not have the important flag set; [...]

Reply:
Yes, I believe that works 👍